Docker Testing - Newer versions of phantomjs and casperjs, in container wheels.

jmchilton opened the pull request:
We are currently testing against "stable" but very old and unmaintained versions of phantomjs and casperjs. phantomjs folks explicitly state 1.9 is unmaintained and the casperjs README recommends running against master. This updates those dependencies in the docker container for testing.

Additionally, this updates the Docker testing setup to use wheel inside the container. This should speed up fresh install testing and prevent certain classes of problems where the wheels from the galaxy mounted into the container don't match the user, version, etc....

carlfeberhard commented:
I didn't think casper worked with phantom 2.0. It usually said: 'CasperJS needs PhantomJS v1.x'. Is this changed now?

jmchilton commented:
They claim that master has some support for PhantomJS 2.0.

https://github.com/n1k0/casperjs/pull/1310 https://github.com/n1k0/casperjs/pull/1352

And that master is the way to run casperjs. "Important note: the master branch hosts the development version of CasperJS, which is now pretty stable and should be the right version to use if you ask me." - first line of readme.

carlfeberhard commented:
Sounds good - I had seen readme but they hadn't inc'd the beta version and I was wondering if you had tried it yourself.

bgruening commented on test/docker/base/Dockerfile:
```diff
@@ -27,9 +27,13 @@ RUN apt-get update -y && apt-get install -y software-properties-common curl && \
-    npm install -g grunt-contrib-qunit grunt grunt-cli casperjs phantomjs && \
     apt-get autoremove -y && apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 
+RUN npm install -g grunt-contrib-qunit grunt grunt-cli
```
I would squeeze this in one RUN commands.

bgruening commented:
@jmchilton do you still have these Docker related problems?

jmchilton committed:
Fix create/manage db scripts to respect GALAXY_VIRTUAL_ENV.

jmchilton committed:
Dockerfile testing improvements.

- Use wheel from inside container.
- Update to the latest pre-release phantom and casper dependencies (they seem more stable).

jmchilton committed:
Update to phantomjs 2.1 and official sources.

jmchilton commented:
Figured out my local problem with Dockerized tests (will have a fix PR opened shortly). Current failures locally for newer versions of phantom and casper.

```
casperjs_runner: DEBUG: (api-history-permission-tests.js): FAIL CurrentHistoryView() [object Object] [object Object] An error was encountered while loading the contents [object Object]
casperjs_runner: DEBUG: (api-history-permission-tests.js): #    type: uncaughtError
casperjs_runner: DEBUG: (api-history-permission-tests.js): #    file: /galaxy/test/casperjs/api-history-permission-tests.js:370
casperjs_runner: DEBUG: (api-history-permission-tests.js): #    error: CurrentHistoryView() [object Object] [object Object] An error was encountered while loading the contents [object Object]
casperjs_runner: DEBUG: (api-history-permission-tests.js): #           _pageErrorHandler@phantomjs://platform/spaceghost.js:370:46
casperjs_runner: DEBUG: (api-history-permission-tests.js): #           emit@phantomjs://platform/events.js:84:23
casperjs_runner: DEBUG: (api-history-permission-tests.js): #           onError@phantomjs://platform/casper.js:2530:20
casperjs_runner: DEBUG: (api-history-permission-tests.js): #           connector@phantomjs://platform/bootstrap.js:60:22
casperjs_runner: DEBUG: (api-history-permission-tests.js): #           error@[native code]
casperjs_runner: DEBUG: (api-history-permission-tests.js): #           http://localhost:9862/static/scripts/bundled/analysis.bundled.js?v=1456748554:7:38251
casperjs_runner: DEBUG: (api-history-permission-tests.js): #           c@http://localhost:9862/static/scripts/bundled/libs.bundled.js?v=1456748554:26:53829
casperjs_runner: DEBUG: (api-history-permission-tests.js): #           trigger@http://localhost:9862/static/scripts/bundled/libs.bundled.js?v=1456748554:26:52962
casperjs_runner: DEBUG: (api-history-permission-tests.js): #           http://localhost:9862/static/scripts/bundled/analysis.bundled.js?v=1456748554:10:28661
casperjs_runner: DEBUG: (api-history-permission-tests.js): #           c@http://localhost:9862/static/scripts/bundled/libs.bundled.js?v=1456748554:24:22528
casperjs_runner: DEBUG: (api-history-permission-tests.js): #           fireWith@http://localhost:9862/static/scripts/bundled/libs.bundled.js?v=1456748554:24:45695
casperjs_runner: DEBUG: (api-history-permission-tests.js): #           http://localhost:9862/static/scripts/bundled/libs.bundled.js?v=1456748554:24:47807
casperjs_runner: DEBUG: (api-history-permission-tests.js): #           http://localhost:9862/static/scripts/bundled/analysis.bundled.js?v=1456748554:7:40913
casperjs_runner: DEBUG: (api-history-permission-tests.js): #           c@http://localhost:9862/static/scripts/bundled/libs.bundled.js?v=1456748554:24:22528
casperjs_runner: DEBUG: (api-history-permission-tests.js): #           fireWith@http://localhost:9862/static/scripts/bundled/libs.bundled.js?v=1456748554:24:45695
casperjs_runner: DEBUG: (api-history-permission-tests.js): #           http://localhost:9862/static/scripts/bundled/libs.bundled.js?v=1456748554:24:47807
casperjs_runner: DEBUG: (api-history-permission-tests.js): #           c@http://localhost:9862/static/scripts/bundled/libs.bundled.js?v=1456748554:24:22528
casperjs_runner: DEBUG: (api-history-permission-tests.js): #           fireWith@http://localhost:9862/static/scripts/bundled/libs.bundled.js?v=1456748554:24:45695
casperjs_runner: DEBUG: (api-history-permission-tests.js): #           i@http://localhost:9862/static/scripts/bundled/libs.bundled.js?v=1456748554:26:5362
casperjs_runner: DEBUG: (api-history-permission-tests.js): #           e@http://localhost:9862/static/scripts/bundled/libs.bundled.js?v=1456748554:26:11538
casperjs_runner: DEBUG: (api-history-permission-tests.js): #           openUrl@[native code]
casperjs_runner: DEBUG: (api-history-permission-tests.js): #           open@phantomjs://platform/casper.js:1453:22
casperjs_runner: DEBUG: (api-history-permission-tests.js): #           open@phantomjs://platform/spaceghost.js:485:39
casperjs_runner: DEBUG: (api-history-permission-tests.js): #           _step@phantomjs://platform/casper.js:1862:18
casperjs_runner: DEBUG: (api-history-permission-tests.js): #           runStep@phantomjs://platform/casper.js:1583:31
casperjs_runner: DEBUG: (api-history-permission-tests.js): #           checkStep@phantomjs://platform/casper.js:404:28
casperjs_runner: DEBUG: (api-history-permission-tests.js): #    stack: not provided
```

```
casperjs_runner: DEBUG: beginning headless browser tests: history-panel-tests.js
casperjs_runner: DEBUG: process_command_list: ['casperjs', 'test', '/galaxy/test/casperjs/history-panel-tests.js', '--url=http://localhost:9862', '--return-json', '--data={}']
casperjs_runner: DEBUG: (history-panel-tests.js): # Testing the form of the main/current history panel
casperjs_runner: DEBUG: (history-panel-tests.js): PASS Testing the form of the main/current history panel (NaN test)
galaxy.webapps.galaxy.controllers.user: DEBUG: trans.app.config.auth_config_file: ./config/auth_conf.xml.sample
casperjs_runner: DEBUG: (history-panel-tests.js): FAIL CurrentHistoryView() [object Object] [object Object] An error was encountered while loading the history [object Object]
casperjs_runner: DEBUG: (history-panel-tests.js): #    type: uncaughtError
casperjs_runner: DEBUG: (history-panel-tests.js): #    file: /galaxy/test/casperjs/history-panel-tests.js:370
casperjs_runner: DEBUG: (history-panel-tests.js): #    error: CurrentHistoryView() [object Object] [object Object] An error was encountered while loading the history [object Object]
casperjs_runner: DEBUG: (history-panel-tests.js): #           _pageErrorHandler@phantomjs://platform/spaceghost.js:370:46
casperjs_runner: DEBUG: (history-panel-tests.js): #           emit@phantomjs://platform/events.js:84:23
casperjs_runner: DEBUG: (history-panel-tests.js): #           onError@phantomjs://platform/casper.js:2530:20
casperjs_runner: DEBUG: (history-panel-tests.js): #           connector@phantomjs://platform/bootstrap.js:60:22
casperjs_runner: DEBUG: (history-panel-tests.js): #           error@[native code]
casperjs_runner: DEBUG: (history-panel-tests.js): #           http://localhost:9862/static/scripts/bundled/analysis.bundled.js?v=1456748554:7:38251
casperjs_runner: DEBUG: (history-panel-tests.js): #           c@http://localhost:9862/static/scripts/bundled/libs.bundled.js?v=1456748554:26:53829
casperjs_runner: DEBUG: (history-panel-tests.js): #           trigger@http://localhost:9862/static/scripts/bundled/libs.bundled.js?v=1456748554:26:52962
casperjs_runner: DEBUG: (history-panel-tests.js): #           http://localhost:9862/static/scripts/bundled/analysis.bundled.js?v=1456748554:10:28661
casperjs_runner: DEBUG: (history-panel-tests.js): #           c@http://localhost:9862/static/scripts/bundled/libs.bundled.js?v=1456748554:24:22528
casperjs_runner: DEBUG: (history-panel-tests.js): #           fireWith@http://localhost:9862/static/scripts/bundled/libs.bundled.js?v=1456748554:24:45695
casperjs_runner: DEBUG: (history-panel-tests.js): #           http://localhost:9862/static/scripts/bundled/libs.bundled.js?v=1456748554:24:47807
casperjs_runner: DEBUG: (history-panel-tests.js): #           http://localhost:9862/static/scripts/bundled/analysis.bundled.js?v=1456748554:7:40721
casperjs_runner: DEBUG: (history-panel-tests.js): #           c@http://localhost:9862/static/scripts/bundled/libs.bundled.js?v=1456748554:24:22528
casperjs_runner: DEBUG: (history-panel-tests.js): #           fireWith@http://localhost:9862/static/scripts/bundled/libs.bundled.js?v=1456748554:24:45695
casperjs_runner: DEBUG: (history-panel-tests.js): #           i@http://localhost:9862/static/scripts/bundled/libs.bundled.js?v=1456748554:26:5362
casperjs_runner: DEBUG: (history-panel-tests.js): #           e@http://localhost:9862/static/scripts/bundled/libs.bundled.js?v=1456748554:26:11538
casperjs_runner: DEBUG: (history-panel-tests.js): #           openUrl@[native code]
casperjs_runner: DEBUG: (history-panel-tests.js): #           open@phantomjs://platform/casper.js:1455:9
casperjs_runner: DEBUG: (history-panel-tests.js): #           open@phantomjs://platform/spaceghost.js:485:39
casperjs_runner: DEBUG: (history-panel-tests.js): #           _step@phantomjs://platform/casper.js:1862:18
casperjs_runner: DEBUG: (history-panel-tests.js): #           runStep@phantomjs://platform/casper.js:1583:31
casperjs_runner: DEBUG: (history-panel-tests.js): #           checkStep@phantomjs://platform/casper.js:404:28
casperjs_runner: DEBUG: (history-panel-tests.js): #    stack: not provided
casperjs_runner: DEBUG: (history-panel-tests.js): # history panel with a new, empty history should be well formed
casperjs_runner: DEBUG: (history-panel-tests.js): # history name should exist, be visible, and have text Unnamed history
```

jmchilton commented:
Taking this out of WIP - the above problems are probably what use to cause phantomjs to crash outright and the individual tests seem to run properly if executed by themselves.

jmchilton commented:
Nevermind, moving back into WIP.

jmchilton commented:
Taking back out of WIP - there are still transient errors but there were before under my setup.

martenson commented:
worth a shot, we need a new docker image, right @jmchilton ?
